code/game/objects/obj_defense.dm

```
@@ -4,9 +4,12 @@
 	if(QDELETED(src))
 		stack_trace("[src] taking damage after deletion")
 		return
+	if(obj_integrity <= 0)
```

So stuff on the griddle doesnt silently error